src/Facilitator.ts

```diff
@@ -9,29 +9,49 @@ export default class Facilitator {
 
   private auxiliarySubscriber: Subscriber;
 
+  private subscriptionRestartTime: number;
+
   /**
    * @param originSubscriber Origin subscriber instance.
    * @param auxiliarySubscriber Auxiliary subscriber instance.
    */
   public constructor(originSubscriber: Subscriber, auxiliarySubscriber: Subscriber) {
     this.originSubscriber = originSubscriber;
     this.auxiliarySubscriber = auxiliarySubscriber;
+    this.subscriptionRestartTime = 10*60*1000; // 10 minutes
   }
 
   /** Starts the facilitator by subscribing to subscription queries. */
   public async start(): Promise<void> {
-    Logger.info('Starting subscription to block chain events');
-    await this.originSubscriber.subscribe();
-    Logger.info('Subscription to origin block chain is done');
-    await this.auxiliarySubscriber.subscribe();
-    Logger.info('Subscription to auxiliary block chain is done');
+    this.subscribeToSubGraphs();
+    setInterval(() => this.restartSubscription(), this.subscriptionRestartTime);
```

Comment:
Please clear the interval when facilitator stops.

Reply:
Done!